flash/tabular/classification/data/data.py

```diff
@@ -82,6 +82,15 @@ def __init__(
         dfs = [train_df]
         self._test_df = None
 
+        if categorical_input is None and numerical_input is None:
+            raise TypeError('Both categorical_input and numerical_input are None!')
```

Comment:
It's not `TypeError`, as the argument received is of the appropriate type. You could have `from pytorch_lightning.utilities.exceptions import MisconfigurationException` or `ValueError` instead.